### PR TITLE
Don't mutate displayOrder if we can't find the page we're moving.

### PR DESCRIPTION
**Pull request by rnewman**

### CODE REVIEW

**Comment by mozilla-licensing** [comment]
Hi there! Thanks for your pull request. This is a reminder that unless you have a separate agreement with Mozilla, your code  [Submissions](https://www.mozilla.org/en-US/about/legal/terms/mozilla/#446-content-submissions) to this Mozilla open source project are made under the terms of the corresponding license of the particular open source project to which you are contributing. For more information on the specific license, please see the LICENSE file in the corresponding repository.

If you did not write the code that you are Submitting, please help us by providing information described in our [Commit Access Requirements](https://www.mozilla.org/en-US/about/governance/policies/commit/requirements/) so that a reviewer can conduct diligence.

If you don't want to see this reminder again, you can respond 'I understand'.

**Comment by rnewman** [comment]
Extended still further, @jsantell. r?

File: app/ui/browser-modern/reducers/pages.js [hunk 1]
```diff
@@ -78,8 +78,13 @@ function createPage(state, id, location = UIConstants.HOME_PAGE, options = {
 
 function removePage(state, pageId) {
   return state.withMutations(mut => {
-    const pageCount = state.displayOrder.size;
     const pageIndex = state.displayOrder.findIndex(id => id === pageId);
+    if (pageIndex === -1) {
+      logger.warn(`Removing page ${pageId} that's not present in displayOrder.`);
```
Comment (jsantell) [inline app/ui/browser-modern/reducers/pages.js]: We should probably just make this a thrown error instead

Comment (victorporof) [inline app/ui/browser-modern/reducers/pages.js]: When can this happen?

**Comment by rnewman** [comment]
Landed with @jsantell's suggestion to throw in both cases.

48284dc

**Pull request closed without merging**